buildSrc/src/main/java/com/google/firebase/gradle/plugins/PublishingPlugin.kt

```diff
@@ -115,45 +122,60 @@ abstract class PublishingPlugin : Plugin<Project> {
    *
-   * While [ReleaseConfig] expects a certain format for its libraries (see [ReleaseConfig.toFile]),
-   * the `projectsToRelease` property has two ways of structuring libraries;
-   * - The artifactId: `firebase-appcheck-debug`
-   * - The project path: `:appcheck:firebase-appcheck-debug`
+   * The [ReleaseConfig] is a pre-defined set of data for a release. It expects a valid [file]
+   * [ReleaseConfig.fromFile], which provides a list of libraries via their [path]
+   * [FirebaseLibraryExtension.getPath]. Additionally, it does __NOT__ take into account
+   * co-releasing libraries-> meaning libraries that should be releasing alongside one another will
+   * need to be individually specified in the [ReleaseConfig], otherwise it will likely cause an
+   * error during the release process.
    *
-   * Either option can take any number of arguments via a comma seperated list:
+   * Example usage of `projectsToPublish`:
    * ```
-   * ./gradlew firebasePublish -PprojectsToRelease="firebase-firestore,firebase-common"
+   * ./gradlew firebasePublish -PprojectsToPublish="firebase-firestore,firebase-common"
    * ```
    *
-   * If both the `projectsToRelease` and [ReleaseConfig] are either empty, or do not exist- this
-   * method will return an empty list.
+   * See [ReleaseConfig.toFile] for example usage of [ReleaseConfig].
+   *
+   * If both `projectsToPublish` and [ReleaseConfig] are empty, or do not exist- this method will
+   * return an empty list.
    */
-  private fun computeReleasingLibraries(project: Project): List<FirebaseLibraryExtension> {
-    val releasingLibs = fetchReleasingLibs(project)
-    val allFirebaseLibraries = project.subprojects.mapNotNull { it.firebaseLibraryOrNull }
+  private fun computeReleasingLibraries(
+    project: Project,
+    allFirebaseLibraries: List<FirebaseLibraryExtension>
+  ): List<FirebaseLibraryExtension> {
+    val releasingLibrariesFromProperty = releasingLibsFromProperty(project)
+    if (releasingLibrariesFromProperty != null) {
+      return allFirebaseLibraries
+        .filter { it.artifactId.get() in releasingLibrariesFromProperty }
+        .flatMap { it.librariesToRelease }
+        .distinctBy { it.artifactId.get() }
+    }
 
-    return allFirebaseLibraries.filter {
-      it.artifactId.get() in releasingLibs || it.path in releasingLibs
+    val releasingLibrariesFromReleseConfig = releasingLibsFromReleaseConfig(project)
+    if (releasingLibrariesFromReleseConfig != null) {
+      return allFirebaseLibraries.filter { it.path in releasingLibrariesFromReleseConfig }
     }
-  }
 
-  private fun fetchReleasingLibs(project: Project) =
-    releasingLibsFromProperty(project) ?: releasingLibsFromReleaseConfig(project) ?: emptyList()
+    return emptyList()
+  }
 
   private fun releasingLibsFromProperty(project: Project) =
-    project.provideProperty<String>("projectsToRelease").orNull?.split(",")
+    project.provideProperty<String>("projectsToPublish").orNull?.split(",")
```

Comment:
What about 

```
project.provideProperty<String>("projectsToPublish").orNull?.split(",") ?: emptyList()
```

Then you don't have to check for nullability outside

Reply:
We want the nullability; that way we can recognize situations where `projectsToPublish` is provided, but set to an empty list. Additionally, `releasingLibsFromReleaseConfig` also returns null. The change would just end up being an `isEmpty()` check versus `!= null`, as it's only used in one place; the function just serves as a means of separation of concerns and readability, not necessarily to be exposed + reused (where nullability would be a higher concern).